### ISSUE TITLE
Avoid checking duplicates

### ISSUE THREAD

**Issue by dcastro** [serokell/xrefcheck#74]
# Clarification and motivation

Say we have a file that contains multiple references to the same link:

```md
Some text and [a link](http://google.com).
More text and [another link](http://google.com).
```

In situations like these, it would be wasteful to check the same link twice.

We should avoid verifying links we've already checked..

If one reference is broken, `xrefcheck` should still report _both_ lines, individually.

PS: This, of course, also applies to when we have multiple files referencing the same link.

# Acceptance criteria

* When a link appears twice or more in a file (or a set of files), we only verify the link once.
* When the link is invalid, we still report each reference individually.
